src/Engine/Graphics/Outdoor.cpp

```diff
@@ -2121,6 +2119,13 @@ void ODM_ProcessPartyActions() {
         pParty->pos.y == partyNewPos.y && pParty->pos.z == partyNewPos.z) {
         if (((pParty->pos.z <= newGroundLevel || partyHasHitModel) && savedZ < 0)) {
             pParty->velocity.z = 0;
+
+            if (partyIsOnWater && savedZ < -400.0f) {
```

Comment:
Why the check for savedZ?

Reply:
Party should have a bit of impact into a water surface to create a splash. Stops it being too spammy.

Reply:
Shouldn't it be dt dependent then? On high fps you might never reach -400

Reply:
Vertical speed accumulates in line with dt so this will work at any fps

Reply:
But it is a magic number that needs a comment :)